packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts

```diff
@@ -0,0 +1,290 @@
+		version: 1,
+		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
+		description: 'Consume Google Tasks API.',
+		defaults: {
+			name: 'Google Tasks',
+			color: '#3E87E4'
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'googleTasksOAuth2Api',
+				required: true
+			}
+		],
+		properties: [
+			{
+				displayName: 'Resource',
+				name: 'resource',
+				type: 'options',
+				options: [
+					{
+						name: 'Task',
+						value: 'task'
+					}
+				],
+				default: 'task',
+				description: 'The resource to operate on.'
+			},
+			...taskOperations,
+			...taskFields
+		]
+	};
+	methods = {
+		loadOptions: {
+			// Get all the tasklists to display them to user so that he can select them easily
+
+			async getTasks(
+				this: ILoadOptionsFunctions
+			): Promise<INodePropertyOptions[]> {
+				const returnData: INodePropertyOptions[] = [];
+				const tasks = await googleApiRequestAllItems.call(
+					this,
+					'items',
+					'GET',
+					'/tasks/v1/users/@me/lists'
+				);
+				for (const task of tasks) {
+					const taskName = task.title;
+					const taskId = task.id;
+					returnData.push({
+						name: taskName,
+						value: taskId
+					});
+				}
+				return returnData;
+			}
+		}
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+		const length = (items.length as unknown) as number;
+		const qs: IDataObject = {};
+		let responseData;
+		const resource = this.getNodeParameter('resource', 0) as string;
+		const operation = this.getNodeParameter('operation', 0) as string;
+		let body: IDataObject = {};
+		for (let i = 0; i < length; i++) {
+			if (resource === 'task') {
+				if (operation === 'create') {
+					body = {};
+					//https://developers.google.com/tasks/v1/reference/tasks/insert
+					const taskId = this.getNodeParameter('task', i) as string;
+					const additionalFields = this.getNodeParameter(
+						'additionalFields',
+						i
+					) as IDataObject;
+
+					if (additionalFields.parent)
+						qs.parent = additionalFields.parent as string;
+
+					if (additionalFields.previous)
+						qs.previous = additionalFields.previous as string;
+
+					if (additionalFields.links) {
+						body.links = (additionalFields.links as string[]).map(link => {
+							return { link: link };
+						});
+					}
+					if (additionalFields.status)
+						body.status = additionalFields.status as string;
+
+					if (additionalFields.notes)
+						body.notes = additionalFields.notes as string;
+
+					if (additionalFields.title)
+						body.title = additionalFields.title as string;
+
+					if (additionalFields.dueDate)
+						body.dueDate = additionalFields.dueDate as string;
+
+					if (additionalFields.completed)
+						body.completed = additionalFields.completed as string;
+
+					if (additionalFields.deleted)
+						body.deleted = additionalFields.deleted as boolean;
```

Comment:
Use brackets `{}` or shorthand `additionalFields.parent ? qs.parent = additionalFields.parent as string;`

See here:
https://stackoverflow.com/a/4797293/1757060

Reply:
okay, I ll do that!!